src/fermilib/transforms/_bravyi_kitaev.py

```diff
@@ -37,53 +41,102 @@ def bravyi_kitaev(operator, n_qubits=None):
-                         fenwick_tree.get_update_set(index)]
-
-            # The C(j) set.
-            ancestor_children = [node.index for node in
-                                 fenwick_tree.get_remainder_set(index)]
-
-            # Switch between lowering/raising operators.
-            d_coefficient = .5j
-            if ladder_operator[1]:
-                d_coefficient *= -1.
-
-            # The fermion lowering operator is given by
-            # a = (c+id)/2 where c, d are the majoranas.
-            d_majorana_component = QubitOperator(
-                (((ladder_operator[0], 'Y'),) +
-                 tuple((index, 'Z') for index in ancestor_children) +
-                 tuple((index, 'X') for index in ancestors)),
-                d_coefficient)
-
-            c_majorana_component = QubitOperator(
-                (((ladder_operator[0], 'X'),) +
-                 tuple((index, 'Z') for index in parity_set) +
-                 tuple((index, 'X') for index in ancestors)),
-                0.5)
-
-            # Update term.
-            transformed_term *= c_majorana_component + d_majorana_component
-        transformed_operator += transformed_term
-    return transformed_operator
+    transformed_terms = (
+        _transform_operator_term(term=term,
+                                 coefficient=operator.terms[term],
+                                 fenwick_tree=fenwick_tree)
+        for term in operator.terms
+    )
+    return inline_sum(seed=QubitOperator(), summands=transformed_terms)
+
+
+def _transform_operator_term(term, coefficient, fenwick_tree):
+    """
+    Args:
+        term (list[tuple[int, int]]):
+            A list of (mode, raising-vs-lowering) ladder operator terms.
+        coefficient (float):
+        fenwick_tree (FenwickTree):
+    Returns:
+        QubitOperator:
+    """
+
+    # Build the Bravyi-Kitaev transformed operators.
+    transformed_ladder_ops = (
+        _transform_ladder_operator(ladder_operator, fenwick_tree)
+        for ladder_operator in term
+    )
+    return inline_product(seed=QubitOperator((), coefficient),
+                          factors=transformed_ladder_ops)
+
+
+def _transform_ladder_operator(ladder_operator, fenwick_tree):
+    """
+    Args:
+        ladder_operator (tuple[int, int]):
+        fenwick_tree (FenwickTree):
+    Returns:
+        QubitOperator:
+    """
+    index = ladder_operator[0]
+
+    # Parity set. Set of nodes to apply Z to.
+    parity_set = [node.index for node in
+                  fenwick_tree.get_parity_set(index)]
+
+    # Update set. Set of ancestors to apply X to.
+    ancestors = [node.index for node in
+                 fenwick_tree.get_update_set(index)]
+
+    # The C(j) set.
+    ancestor_children = [node.index for node in
+                         fenwick_tree.get_remainder_set(index)]
+
+    # Switch between lowering/raising operators.
+    d_coefficient = -.5j if ladder_operator[1] else .5j
```

Comment:
The reason for this is because of the notation used by physicists. The notation is that a^\dagger raises and a lowers. If the notation was a^\dagger and a^- or something like that, we would have done -1 and 1.